### PR TITLE
Complete unittest for trainer_config_helpers.

### PR DESCRIPTION
**Pull request by reyoung**

### CODE REVIEW

File: python/paddle/trainer_config_helpers/tests/configs/check.md5 [hunk 1]
```diff
@@ -0,0 +1,3 @@
+9c038249ec8ff719753a746cdb04c026  layer_activations.protostr
```
Comment (reyoung) [inline python/paddle/trainer_config_helpers/tests/configs/check.md5]: @emailweixu I just generate proto string and check the md5sum same.

I read the proto string manually to check proto string correctly. Is this way ok?

**Comment by reyoung** [comment]
`layers.py` unittest complete.

File: python/paddle/trainer_config_helpers/layers.py [hunk 2]
```diff
@@ -2150,35 +2254,44 @@ def get_output_layer(input, arg_name, name=None, layer_attr=None):
 @wrap_param_attr_default()
 @layer_support()
 def recurrent_layer(input, act=None, bias_attr=None,
-                    param_attr=None, name=None, layer_attr=None):
+                    param_attr=None, name=None, reverse=False, layer_attr=None):
     """
-    TODO(yuyang18): Add docs
+    Simple recurrent unit layer. It is just a fully connect layer through both
+    time and neural network.
```
Comment (luotao1) [inline python/paddle/trainer_config_helpers/layers.py]: The note can copy formula from recurrent_layer.

File: python/paddle/trainer_config_helpers/layers.py [hunk 3]
```diff
@@ -2802,18 +2939,20 @@ def tensor_layer(input, size, act=None, name=None,
 
     .. code-block:: python
 
-       tensor = tensor_layer(input=[layer1, layer2])
+       tensor = tensor_layer(x1=layer1, x2=layer2, size=1000)
```
Comment (luotao1) [inline python/paddle/trainer_config_helpers/layers.py]: the input of tensor_layer is x1 and x2, conv_shift_layer is a and b, and the dotmul_operator is x and y, can make them unifined?

Comment (luotao1) [inline python/paddle/trainer_config_helpers/layers.py]: the input of tensor_layer is x1 and x2, conv_shift_layer is a and b, and the dotmul_operator is x and y, can make them unifined?

File: python/paddle/trainer_config_helpers/tests/configs/beam_search.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+beam_gen = beam_search(name="decoder",
+                       step=rnn_step,
+                       input=[StaticInput(input=dat), GeneratedInput(
+                           embedding_name='emb', embedding_size=128, size=128
+                       )],
+                       bos_id=0,
+                       eos_id=1,
+                       beam_size=5,
+                       result_file="./generated_sequences.txt")
```
Comment (luotao1) [inline python/paddle/trainer_config_helpers/tests/configs/beam_search.py]: there has unitest (sample_trainer_rnn_gen.conf) for beam search, and the api is refined in pull#122

File: python/paddle/trainer_config_helpers/layers.py [hunk 5]
```diff
@@ -762,16 +786,20 @@ def print_layer(input, name=None):
...
+    for each in input:
+        assert isinstance(each, LayerOutput)
 
     Layer(
         name=name,
         type=LayerType.PRINT_LAYER,
         inputs=[l.name for l in input],
     )
-    LayerOutput(name, LayerType.PRINT_LAYER, input)
+    return LayerOutput(name, LayerType.PRINT_LAYER, input)
```
Comment (emailweixu) [inline python/paddle/trainer_config_helpers/layers.py]: print_layer should not return anything

**Pull request merged**